cmake/checks/check_01_cxx_features.cmake

```diff
@@ -646,7 +647,7 @@ else()
     DEAL_II_CXX14_CONSTEXPR_BUG)
 endif()
 
-set(DEAL_II_CONSTEXPR "constexpr")
+set(DEAL_II_CONSTEXPR "constexpr __host__")
```

Comment:
This is in a generic cmake section, unrelated to CUDA or Kokkos or other stuff. But `__host__` is, I assume, not something compilers will understand if compiler for a regular CPU. Is this what you wanted?

Reply:
Yes, this would only be if compiling with a compiler that understands `CUDA` or `HIP`.
At this point, my question mostly is if we care enough about supporting `Clang+Cuda` to make all these changes.